test/compilable/test23089.d

```diff
@@ -0,0 +1,7 @@
+// https://issues.dlang.org/show_bug.cgi?id=23089
```

Comment:
You can use the trait getLinkage here to verify the applied linkage.

Reply:
It's a backend ICE, not a front-end.